front/src/applications/stdcm/utils/consistValidation.ts

```diff
@@ -0,0 +1,88 @@
+import { kgToT } from 'utils/physics';
+
+export const CONSIST_TOTAL_MASS_MAX = 10000; // ton
+export const CONSIST_TOTAL_LENGTH_MAX = 750; // m
+export const CONSIST_MAX_SPEED_MIN = 30; // km/h
+
+export const validateTotalMass = ({
+  tractionEngineMass = 0,
+  towedMass = 0,
+  totalMass,
+}: {
+  tractionEngineMass?: number;
+  towedMass?: number;
+  totalMass?: number;
+}) => {
+  if (!totalMass) {
+    return undefined;
+  }
+
+  if (totalMass <= 0) {
+    return 'consist.errors.totalMass.negative';
+  }
+
+  const tractionMassInTons = kgToT(tractionEngineMass);
+  const consistMassInTons = kgToT(tractionEngineMass + towedMass);
+  const massLimit = towedMass ? consistMassInTons : tractionMassInTons;
+
+  if (totalMass < massLimit || totalMass >= CONSIST_TOTAL_MASS_MAX) {
+    return 'consist.errors.totalMass.range';
+  }
+
+  if (
+    totalMass < (towedMass ? consistMassInTons : tractionMassInTons) ||
+    totalMass >= CONSIST_TOTAL_MASS_MAX
+  ) {
+    return 'consist.errors.totalMass.range';
+  }
+
+  return undefined;
+};
+
+export const validateTotalLength = ({
+  tractionEngineLength = 0,
+  towedLength = 0,
+  totalLength,
+}: {
+  tractionEngineLength?: number;
+  towedLength?: number;
+  totalLength?: number;
+}) => {
+  if (!totalLength) {
+    return undefined;
+  }
+
+  if (totalLength <= 0) {
+    return 'consist.errors.totalLength.negative';
+  }
+
+  const consistLength = tractionEngineLength + towedLength;
+
+  if (
+    totalLength < Math.max(tractionEngineLength, consistLength) ||
```

Comment:
Can `tractionEngineLength` ever be lower than `consistLength`? I think these are all positive numbers right?